docs/proposals/operator-sdk-v1-migration.md

```diff
@@ -0,0 +1,285 @@
+
+A few of the common development commands have been moved out of operator-sdk:
+
+1. `operator-sdk generate crds` has been replaced with the generated `make manifests` target.
+2. `operator-sdk build` has likewise been replaced with a make target - `make docker-build` [13].
+3. Deploying is likewise handled by a make targets - `make install` and `make deploy` [14].
+
+The generated Makefile targets can be used by contributors to test Shipwright Builds. We are free
+to add our own targets to generate an "all-in-one" YAML manifest.
+
+[13] https://sdk.operatorframework.io/docs/building-operators/golang/migration/#generate-manifests-and-build-the-operator
+[14] https://sdk.operatorframework.io/docs/building-operators/golang/quickstart/
+
+
+### Risks and Mitigations
+
+**Risk**: Bug fixes and features are wiped out as we migrate to operator-sdk v1.
+
+*Mitigation*: As the rebase PR nears completion, we hold outstanding pull requests until it merges.
+Our CI needs to ensure that the rebase can merge without conflicts.
+
+## Design Details
+
+### Test Plan
+
+Existing test suites would remain in place, with overhauls to utilize envtest for integration testing.
+The generated `make test` target runs `go test ./...`, 
+
+### Graduation Criteria
+
+Not applicable
+
+### Upgrade / Downgrade Strategy
+
+Since this project is in the alpha state, support for in-place upgrade is not a hard requirement.
+This proposal outlines specific areas where there is potential to introduce breaking changes that
+downstream distributions may need to consider.
+
+### Version Skew Strategy
+
+If the leader election ConfigMap is left unchanged, we can support operator skew.
+
+## Implementation History
+
+2020-12-08: Initial proposal
+2020-12-09: Alternative using knative/pkg
+
+## Drawbacks
+
+Migrating to operator-sdk v1 requires significant project restructuring and changes to how we run
+tests. Bug fixes and other enhancements could easily get lost in the shuffle.
+
+The primary benefit of operator-sdk is to facilitate the deployment of Shipwright Build as an
+operator managed by the Operator Lifecycle Manager. operator-sdk has tooling to help create the
+necessary artifacts for OLM, but one does not need operator-sdk to create the components needed to
+deploy an operator via OLM.
+
+## Alternatives
+
+### Use Upstream Components
+
+We can use the sub-components that operator-sdk packages together, without using operator-sdk
+directly:
+
+1. Controller-runtime can be used to write the controllers and controller manager [15].
+2. KubeBuilder can be used to create the custom resources [16].
+3. Kustomize can be used to manage our YAML manifests [17].
+
+[15] https://github.com/kubernetes-sigs/controller-runtime
+[16] https://github.com/kubernetes-sigs/kubebuilder
+[17] https://kustomize.io/
+
+### Migrate code only
+
+In this scenario, we migrate our CRDs and controller code only. Integration and e2e tests remain
+as-is, potentially move to envtest at a later date.
```

Comment:
I like this alternative. Re-writing our tests is _a lot of pain with no immediate gain_

Reply:
( Though I need to read further if this is possible, with `operator-sdk test` gone.. )

Reply:
I just spend like 1 hour reading all of the above, to realize that at the very end you had the `alternative` I had in mind. Yes, I agree this is ideal, as I mentioned in many comments above, I honestly do not see the benefit of such a big refactoring.